samcli/cli/context.py

```diff
@@ -40,6 +42,10 @@ def __init__(self):
     def debug(self):
         return self._debug
 
+    @property
+    def output_timestamp(self):
```

Comment:
Can't we just configure the logger to have the timestamp when debugging is enabled? Do we really need a new options for this?

Reply:
Related comment from @awsjeffg 

https://github.com/aws/aws-sam-cli/issues/1773#issuecomment-695076282

I think the idea is, maybe, for non-debug mode, users will want still to see the timestamp. So having a dedicate flag gives such flexibility.

Reply:
@jfuss You're not wrong, I am just being careful as we've run into other problems where changing stdout behaviour causes customers problems.

Reply:
@awsjeffg Yes, we have run into problems with other tools expecting certain output but that has been against our 'info' logs not 'debug'. Additionally, we should be able to freely add data without worrying. I see a stronger case for removing  output. I am not saying we shouldn't be cautious around changing output but as a product we need to have the flexibility to update and change output, otherwise we are always stuck with whatever we ship with. I am a strong believer in relying on stdout or stderr output is an anti-pattern and should be treated as behavior not an interface. I am only aware of the AWS Toolkit depending on our output, can't we just ask them if this would break them? My understanding is it shouldn't.

Long ago, we had the default logger for all of SAM CLI's logs. This was overly noisy and we ended up removing it. I personally do not see a difference between adding `--time-stamp-output` and using `--debug`. They are ways to configure the logging and output. We should be aware of the additional overhead on our customer when adding new options for things. Why should we have an option versus an environment variable? If you want this still to be intentionally, wouldn't something like allowing `--debug --debug` to allow customers to control/add additional output (similar to `-vv` which is common for command line tools)? 

I can understand wanting to introduce this in a new way somehow (though I don't fully agree), but the main thing I want to poke at is: Why a new command line option?

Reply:
Both are valid in my opinion. Another thought against **ONLY** including timestamp in DEBUG mode: including timestamp into `--debug` which is used for configuring logging level is implicit in my opinion. 

Ideally I would prefer to always include timestamps regardless what logging level users chooses to configure. But as @awsjeffg pointed out, it will change stdout behaviour. I understand changing stdout is not something we like to do very often, but `--output-timetamp` does feel like a temporary solution. 

Considering users only use `--debug` mode for investigating (are they?), I think it is fairly safe to change stdout behavior there assuming debug stdout won't be used for postprcessing. What do you think?

Reply:
As discussed offline with @awsjeffg, we are going with including timestamp when `--debug` is presented